File: Logica2.java
    import java.util.HashMap;
    import java.util.Map;
    
    public class Logica2 {
        public static void main(String[] args) {
            Map<String, String> hermanos = new HashMap<>();
            hermanos.put("Carlos", "Luis");
    
            Map<String, String> hijos = new HashMap<>();
            hijos.put("Carlos", "José");
            hijos.put("Luis", "Ana");
    
            boolean resultado = esTio("Luis", "José", hermanos, hijos);
            String tio = "Luis";
            String sobrino = "José";
    
            if (resultado) {
                System.out.println(tio + " es tio de " + sobrino);
            } else {
                System.out.println(tio + " no es tio de " + sobrino);
            }
        }
    
        // Check if 'tio' is an uncle of 'sobrino' using 'hermanos' and 'hijos' maps
        public static boolean esTio(String tio, String sobrino, Map<String, String> hermanos, Map<String, String> hijos) {
            // Check if 'tio' has a sibling and if that sibling has 'sobrino' as a child
            for (Map.Entry<String, String> entry : hermanos.entrySet()) {
                if (entry.getValue().equals(tio) && hijos.containsKey(entry.getKey()) && hijos.get(entry.getKey()).equals(sobrino)) {
                    return true;
                }
            }
            return false;
        }
    }

File: Logica2.cpp
#include <iostream>
#include <string>
#include <map>

std::map<std::string, std::string> hermanos = {
    {"Carlos", "Luis"}
};

std::map<std::string, std::string> hijos = {
    {"Carlos", "Jose"},
    {"Luis", "Ana"}
};

bool son_hermanos(const std::string& hermano1, const std::string& hermano2) {
    return hermanos[hermano1] == hermano2;
}

bool es_hijo(const std::string& padre, const std::string& hijo) {
    return hijos[padre] == hijo;
}

int main() {
    bool resultado1 = son_hermanos("Carlos", "Luis");
    bool resultado2 = es_hijo("Carlos", "Jose");

    if (resultado1 && resultado2) {
        std::cout << "Luis es tío de Jose" << std::endl;
    } else {
        std::cout << "Luis NO es tío de Jose" << std::endl;
    }

    return 0;
}
